### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by plasne**
This PR includes all documentation for Batcher and AzureSharedResource.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,15 +1,357 @@
...
+
+- __AzureSharedResource__: This rate limiter allows you to reserve a fixed amount of capacity and then share a fixed amount of capacity across multiple processes.
+
+Some other terms will be used throughout...
+
+- __Target__: As Operations are enqueued or marked done in Batcher it updates a Target number which is the total cost Batcher thinks is necessary to process any outstanding Operations. In other words, as Operations are enqueued, the Target grows by the cost of that Operation. When a batch is marked done, the Target is reduced by the cost of all Operations in that batch.
+
+- __Capacity__: The capacity that the rate limiter has been able to procure is available via the Capacity() method or the capacity event.
+
+- __MaxCapacity__: When using a rate limiter the MaxCapacity is the maximum capacity that could even be provided. For AzureSharedResource, this is the total of shared capacity and reserved capacity. For ProvisionedResource, this is the capacity that was provided when New() was called.
```
Comment (cdrewuk) [inline README.md]: The terms "shared capacity" and "reserved capacity" have not really been introduced by this point in the document and I don't think are going to be obvious to everyone.

Comment (plasne) [reply README.md]: I have added those terms now.

File: provisioned-resource.go [hunk 2]
```diff
@@ -29,10 +30,10 @@ func (r *ProvisionedResource) GiveMe(target uint32) {
 }
 
 func (r *ProvisionedResource) Start(ctx context.Context) error {
-	// nothing to do
+	r.emit("capacity", int(r.maxCapacity), nil)
```
Comment (TechnicallyWilliams) [inline provisioned-resource.go:33]: Minor suggestion for keeping this call and any future MaxCapacity() implementation details in sync.
```suggestion
	r.emit("capacity", r.MaxCapacity(), nil)
```

Comment (plasne) [reply provisioned-resource.go:33]: Fair enough, I could make that change.

**Pull request merged**